Simplenote/NoticeFactory.swift

```diff
@@ -0,0 +1,45 @@
+import Foundation
+
+struct NoticeFactory {
+    static func linkCopied() -> Notice {
+        return Notice(message: Messages.copyLink, action: nil)
+    }
+
+    static func publishing() -> Notice {
+        Notice(message: Messages.publishing, action: nil)
+    }
+
+    static func unpublishing() -> Notice {
+        return Notice(message: Messages.unpublishing, action: nil)
+    }
+
+    static func noteTrashed(_ note: Note, onUndo: @escaping ()-> Void) -> Notice {
+        let action = NoticeAction(title: Messages.undo, handler: onUndo)
+        let notice = Notice(message: Messages.trashed, action: action)
+        return notice
```

Comment:
Nipticky: No need to assign `let notice = ` since we're not doing anything with the value